### ISSUE TITLE
Reply to Mastodon user will not showed as reply in Mastodon

### ISSUE THREAD

**Issue by pullopen** [Plume-org/Plume#834]
Plume works fine with the federation between Plume and Mastodon, which is its advantage to other blog platform. However, when a mastodon user comment an article from mastodon, and then if the plume user replies to this comment on plume, on mastodon it will not show as an reply, but as an original toot, with a full @username@mastodon.domain as plain text. It does not form as a complete "mention". 

- **Plume version:** 0.6.1-dev
- **Operating system:** Windows
- **Web Browser:** Firefox

**Comment by KitaitiMakoto** [comment]
I don't know Plumes whole behavior. So I cannot determine this is a bug or what is just not implemented. I need investigate but it might take a while.

**Comment by KitaitiMakoto** [comment]
Can you write actual Plume and Mastodon account?

**Comment by pullopen** [comment]
> 
> 
> Can you write actual Plume and Mastodon accounts?
> 
> Perhaps this is related to https://git.joinplu.me/Plume/Plume/issues/865

Well, for example, the article here: 

https://plume.pullopen.xyz/~/%E5%9B%BE%E4%B9%A6%E9%A6%86%E7%AE%A1%E7%90%86%E6%97%A5%E5%BF%97/plume-%E6%9B%B4-%E6%96%B0-%E8%87%B3-0.5.0-%E5%90%8E-%E7%9A%84-%E6%B5%8B-%E8%AF%95-%E7%BB%93-%E6%9E%9C 

The plume account made an reply to mastodon account @flyover@pullopen.xyz on Plume, but on mastodon it shows like this: 

https://pullopen.xyz/@flyover/105394047400463952 

The reply below did not make a valid "mention" to the mastodon account.

**Comment by KitaitiMakoto** [comment]
Thanks for the explanation. Let me clear. For me, both on Plume and Mastodon, `admin@plume.pullopen.xyz` replies `flyover@pullopen.xyz` that "然而刚夸完站就挂了……".
![](https://gyazo.com/a4288c653636c19dcb9df42935890184.png)
![](https://gyazo.com/1bc89f81c91e0762616a309a678e2d24.png)

Could you explain more details, please?

**Comment by pullopen** [comment]
> 
> 
> Thanks for the explanation. Let me clear. For me, both on Plume and Mastodon, `admin@plume.pullopen.xyz` replies `flyover@pullopen.xyz` that "然而刚夸完站就挂了……".
> ![](https://camo.githubusercontent.com/362f52cf01e95b55a39ce100470c56bd4fb7d3c3443fd366f4a2d1dd4e1d5d96/68747470733a2f2f6779617a6f2e636f6d2f61343238386336353336333663313964636239646634323933353839303138342e706e67)
> ![](https://camo.githubusercontent.com/4da698ddebb3f7f9450a3eabbb7cfe93c9933f373188f09dba7643386aa36f17/68747470733a2f2f6779617a6f2e636f6d2f31626338396638316339316530373632363136613330396136373865326432342e706e67)
> 
> Could you explain more details, please?

Yes, but in mastodon, the `@flyover@pullopen.xyz` is just a plain text. It's not a valid "mention" and you can not get notification from it on mastodon.

**Comment by KitaitiMakoto** [comment]
Ah, I understood. Okay, I will investigate it. Thank you for reporting.

**Comment by KitaitiMakoto** [comment]
At v0.7.0, actually commenting doesn't include mention links: https://git.joinplu.me/Plume/Plume/src/tag/0.7.0/src/routes/comments.rs#L73-L82

```rust

            for ment in mentions {
                Mention::from_activity(
                    &conn,
                    &Mention::build_activity(&conn, &ment)
                        .expect("comments::create: build mention error"),
                    comm.id,
                    false,
                    true,
                )
                .expect("comments::create: mention save error");
            }
```
These mentions should be included in comment activity.

**Comment by KitaitiMakoto** [comment]
I was wrong. Mentions in a comment are attached here: https://git.joinplu.me/Plume/Plume/src/tag/0.7.0/plume-models/src/comments.rs#L135-L139

**Comment by pullopen** [comment]
> I was wrong. Mentions in a comment are attached _here_: https://git.joinplu.me/Plume/Plume/src/tag/0.7.0/plume-models/src/comments.rs#L135-L139

I just tried in v0.7.3-dev. Now the username mentioned in plume will show as a link in mastodon now (which will even have a link preview) rather than plain text. However I still can not receive any notification in mastodon. Anyway it looks better though.
